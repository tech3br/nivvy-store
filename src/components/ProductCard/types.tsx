export type ProductProps = {
  title: string;
  type: string;
  img?: string;
  price: string;
  promotionalPrice?: string;
  favorite?: boolean;
  ribbon?: React.ReactNode;
  discountText?: string;
  onFav?: () => void;
  soldItems: number;
  totalItems: number;
};
