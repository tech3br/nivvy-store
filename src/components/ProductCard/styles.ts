import LinearProgress, { linearProgressClasses } from '@mui/material/LinearProgress';
import styled from 'styled-components';

export const Container = styled.div`
  position: relative;
  /* Auto layout */

  display: flex;
  flex-direction: column;
  align-items: center;

  width: 282px;
  height: auto;

  /* Style */

  background: ${({ theme }) => theme.colors.white};
  /* Outline */

  border: 1px solid ${({ theme }) => theme.colors.lightGray};
  border-radius: 8px;
  cursor: pointer;

  > svg {
    margin-top: 16px;
    margin-right: 16px;
    width: 24px;
    height: 24px;
    right: 0;
    position: absolute;
    z-index: 10;
  }
`;

export const HeaderImage = styled.div`
  position: relative;
  display: flex;
  width: 100%;
  height: 306px;

  background: #e6e8ec;
  border-radius: 8px 8px 0px 0px;

  img {
    border-radius: 8px 8px 0px 0px;
    object-fit: cover;
    object-position: 100% 90%;
  }
`;

export const ProductCardBody = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 176px;
  padding: 16px;
`;

export const ProductTypeDescription = styled.p`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: 100%;
  font-family: "Inter";
  font-style: normal;
  font-weight: 300;
  font-size: 12px;
  line-height: 15px;
  display: flex;
  align-items: center;
  color: #828282;
`;

export const ProductTypeTitle = styled.p`
  font-family: "Inter";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 140%;
  /* or 22px */

  display: flex;
  align-items: center;

  /* Black */

  color: #212121;
  margin-top: 8px;
`;

export const PriceContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  margin: 16px 0px;
  height: 23px;
`;

export const ProductPrice = styled.span`
  display: flex;
  s {
    font-family: "Inter";
    font-style: normal;
    font-weight: 300;
    font-size: 12px;
    line-height: 19px;

    /* Disable */

    color: #828282;
  }

  strong {
    font-family: "Inter";
    font-style: normal;
    font-weight: 600;
    font-size: 18px;
    line-height: 19px;
    /* Black */

    color: #212121;
  }
`;

export const PromotionDiscount = styled.span`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 4px 6px;
  gap: 8px;

  width: 47px;
  height: 23px;

  background: rgba(255, 0, 0, 0.1);
  border-radius: 4px;

  p {
    /* Caption/Medium */

    font-family: "Inter";
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 15px;
    /* identical to box height */

    display: flex;
    align-items: center;
    text-align: center;

    color: #ff0000;
  }
`;

export const ProgressEndPromotion = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;

  p {
    /* Caption/Regular */

    font-family: "Inter";
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 15px;
    /* identical to box height */

    display: flex;
    align-items: center;

    /* Disable */

    color: #828282;
  }
`;

export const BorderLinearProgress = styled(LinearProgress)`
  ${({ theme }) => ({
    borderRadius: 8,
    [`&.${linearProgressClasses.colorPrimary}`]: {
      backgroundColor: theme.colors.white,
      height: "10px",
    },
    [`& .${linearProgressClasses.bar}`]: {
      borderRadius: 5,
      background: `linear-gradient(90deg, #FF512F 0%, #F09819 100%);`,
    },
  })}
`;
