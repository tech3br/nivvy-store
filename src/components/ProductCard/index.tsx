import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
// import Image from "next/image";
import React from "react";
import { FcLike, FcLikePlaceholder } from "react-icons/fc";
import * as S from "./styles";
import { ProductProps } from "./types";

const ProductCard = ({
  img,
  discountText,
  price,
  promotionalPrice,
  title,
  type,
  soldItems,
  totalItems,
  favorite,
}: ProductProps) => {
  const [progress] = React.useState(80);

  return (
    <S.Container>
      {favorite ? <FcLike /> : <FcLikePlaceholder />}
      <S.HeaderImage>
        {/* <Image src={Offer} width={322} height={332} alt={title} /> */}
      </S.HeaderImage>
      <S.ProductCardBody>
        <S.ProductTypeDescription>{type}</S.ProductTypeDescription>
        <S.ProductTypeTitle>{title}</S.ProductTypeTitle>
        <S.PriceContainer>
          <S.ProductPrice>
            <s>{price}</s>
            <strong>{promotionalPrice}</strong>
          </S.ProductPrice>
          {!!discountText && (
            <S.PromotionDiscount>
              <p>- {discountText}</p>
            </S.PromotionDiscount>
          )}
        </S.PriceContainer>
        <S.ProgressEndPromotion>
          <Box sx={{ height: "10px", width: "100%" }}>
            <S.BorderLinearProgress variant="determinate" value={progress} />
          </Box>
          <Typography
            variant="body1"
            style={{ textAlign: "center" }}
          >{`${soldItems}/${totalItems} vendidos`}</Typography>
        </S.ProgressEndPromotion>
      </S.ProductCardBody>
    </S.Container>
  );
};

export default ProductCard;
