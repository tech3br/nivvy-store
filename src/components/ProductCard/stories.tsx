import { Meta, Story } from "@storybook/react/types-6-0";
import ProductCard from ".";
import { ProductProps } from "./types";

export default {
  title: "ProductCard",
  component: ProductCard,
  args: {
    type: "Fitness",
    favorite: true,
    title: "Garrafa TÃ©rmica VIP",
    img: '/public/images/promo/special-offer.png',
    price: "R$ 235,00",
    promotionalPrice: "R$ 200,00",
    soldItems: 80,
    totalItems: 100,
  },
  argTypes: {
    onFav: { action: "clicked" },
    ribbon: { type: "string" },
  },
  parameters: {
    backgrounds: {
      default: "dark",
    },
  },
} as Meta;

export const Default: Story<ProductProps> = (args) => (
  <div style={{ width: "30rem" }}>
    <ProductCard {...args} />
  </div>
);

export const WithDiscount: Story<ProductProps> = (args) => (
  <div style={{ width: "30rem" }}>
    <ProductCard {...args} discountText="50%" />
  </div>
);
