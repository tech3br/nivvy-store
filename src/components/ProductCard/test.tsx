import { render, screen } from "../../utils/test-utils";

import ProductCard from ".";

const props = {
  title: "Garrafa Térmica VIP",
  type: "Fitness",
  price: "R$ 235,00",
  soldItems: 80,
  totalItems: 100,
};

describe("<ProductCard />", () => {
  it("should be render product card", () => {
    const { container } = render(<ProductCard {...props} />);

    // Verifica se o title existe renderizado no componente
    expect(screen.getByText(/Garrafa Térmica VIP/)).toBeInTheDocument();

    // Verifica se o price existe renderizado no componente
    expect(screen.getByText(`R$ 235,00`)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the product card with discountText", () => {
    render(<ProductCard {...props} discountText="-50%" />);

    const discountText = screen.getByText(/-50%/i);

    expect(discountText).toBeInTheDocument();
    expect(discountText).toHaveStyle({ color: "#ff0000" });
    expect(discountText).toHaveStyle({ fontSize: "12px" });
  });
});
