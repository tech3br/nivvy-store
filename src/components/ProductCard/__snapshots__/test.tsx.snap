// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should be render product card 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 282px;
  height: auto;
  background: #FAFAFA;
  border: 1px solid #EAEAEA;
  border-radius: 8px;
  cursor: pointer;
}

.c0 > svg {
  margin-top: 16px;
  margin-right: 16px;
  width: 24px;
  height: 24px;
  right: 0;
  position: absolute;
  z-index: 10;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 306px;
  background: #e6e8ec;
  border-radius: 8px 8px 0px 0px;
}

.c1 img {
  border-radius: 8px 8px 0px 0px;
  object-fit: cover;
  object-position: 100% 90%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 176px;
  padding: 16px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  font-family: "Inter";
  font-style: normal;
  font-weight: 300;
  font-size: 12px;
  line-height: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #828282;
}

.c4 {
  font-family: "Inter";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 140%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #212121;
  margin-top: 8px;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 16px 0px;
  height: 23px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 s {
  font-family: "Inter";
  font-style: normal;
  font-weight: 300;
  font-size: 12px;
  line-height: 19px;
  color: #828282;
}

.c6 strong {
  font-family: "Inter";
  font-style: normal;
  font-weight: 600;
  font-size: 18px;
  line-height: 19px;
  color: #212121;
}

.c7 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 p {
  font-family: "Inter";
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #828282;
}

.c8 {
  border-radius: 8px;
}

.c8.MuiLinearProgress-colorPrimary {
  background-color: #FAFAFA;
  height: 10px;
}

.c8 .MuiLinearProgress-bar {
  border-radius: 5px;
  background: linear-gradient(90deg,#FF512F 0%,#F09819 100%);
}

<div
  class="c0"
>
  <svg
    enable-background="new 0 0 48 48"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    version="1"
    viewBox="0 0 48 48"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M34,9c-4.2,0-7.9,2.1-10,5.4C21.9,11.1,18.2,9,14,9C7.4,9,2,14.4,2,21c0,11.9,22,24,22,24s22-12,22-24 C46,14.4,40.6,9,34,9z"
      fill="#FFCDD2"
    />
  </svg>
  <div
    class="c1"
  >
    <img
      alt="Garrafa Térmica VIP"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      src=""
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
  <div
    class="c2"
  >
    <p
      class="c3"
    >
      Fitness
    </p>
    <p
      class="c4"
    >
      Garrafa Térmica VIP
    </p>
    <div
      class="c5"
    >
      <span
        class="c6"
      >
        <strong>
          R$ 235,00
        </strong>
      </span>
    </div>
    <div
      class="c7"
    >
      <div
        class="MuiBox-root css-11img8u"
      >
        <span
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="80"
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate c8 css-eglki6-MuiLinearProgress-root"
          role="progressbar"
        >
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
            style="transform: translateX(-20%);"
          />
        </span>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        style="text-align: center;"
      >
        80/100 vendidos
      </p>
    </div>
  </div>
</div>
`;
